cli/replicasets/setup.go

```diff
@@ -0,0 +1,304 @@
+	}
+
+	// set failover priority
+	// This step is performed separately because we need to know instances UUIDs
+	// to change failover priority.
+	// Generally, we know all instances UUIDs only after creating replicaset or
+	// joining new instances to the existing one.
+	newTopologyReplicasets, err = setFailoverPriority(conn, replicasetsList, newTopologyReplicasets)
+	if err != nil {
+		return nil, err
+	}
+
+	return newTopologyReplicasets, nil
+}
+
+func createAndUpdateReplicasets(conn net.Conn, replicasetsList *ReplicasetsList, instancesConf *InstancesConf,
+	topologyReplicasets *TopologyReplicasets) (*TopologyReplicasets, error) {
+
+	editReplicasetsOpts := &EditReplicasetsOpts{}
+	for _, replicasetConf := range *replicasetsList {
+		topologyReplicaset := topologyReplicasets.GetByAlias(replicasetConf.Alias)
+
+		if topologyReplicaset == nil {
+			editReplicasetOpts, err := getCreateReplicasetEditReplicasetsOpts(replicasetConf, instancesConf)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to get edit_topology options for creating replicaset: %s", err)
+			}
+			*editReplicasetsOpts = append(*editReplicasetsOpts, editReplicasetOpts)
+		} else {
+			editReplicasetOpts, err := getUpdateReplicasetEditReplicasetsOpts(topologyReplicaset, replicasetConf, instancesConf)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to get edit_topology options for updating replicaset: %s", err)
+			}
+			*editReplicasetsOpts = append(*editReplicasetsOpts, editReplicasetOpts)
+		}
+	}
+
+	newTopologyReplicasets, err := editReplicasets(conn, editReplicasetsOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return newTopologyReplicasets, nil
+}
+
+func setFailoverPriority(conn net.Conn, replicasetsList *ReplicasetsList, topologyReplicasets *TopologyReplicasets) (*TopologyReplicasets, error) {
+	editReplicasetsOpts := EditReplicasetsOpts{}
+
+	for _, replicasetConf := range *replicasetsList {
+		newTopologyReplicaset := topologyReplicasets.GetByAlias(replicasetConf.Alias)
+
+		// set failover priority
+		editReplicasetOpts, err := getSetFailoverPriorityEditReplicasetOpts(replicasetConf.InstanceNames, newTopologyReplicaset)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get edit_topology options for setting failover priority: %s", err)
+		}
+
+		editReplicasetsOpts = append(editReplicasetsOpts, editReplicasetOpts)
+	}
+
+	newTopologyReplicasets, err := editReplicasets(conn, &editReplicasetsOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return newTopologyReplicasets, nil
+}
+
+func logSetupSummary(topologyReplicasets, newTopologyReplicasets *TopologyReplicasets) {
+	for replicasetUUID, newTopologyReplicaset := range *newTopologyReplicasets {
+		replicasetRes := common.Result{
+			ID: newTopologyReplicaset.Alias,
+		}
+
+		if oldTopologyReplicaset, found := (*topologyReplicasets)[replicasetUUID]; !found {
+			replicasetRes.Status = common.ResStatusCreated
+		} else if reflect.DeepEqual(newTopologyReplicaset, oldTopologyReplicaset) {
```

Comment:
If nothing occurred it's OK, am I right? Why not "not modified"?

Reply:
See comment above.